### PR TITLE
Add zng_ prefix to internal functions to avoid linking conflicts with zlib.

### PR DESCRIPTION
**Pull request by nmoinvaz**
This pull requests adds zng_ prefix to certain internal functions to allow for dual linking both zlib & zlib-ng. #358, #360

### CODE REVIEW

File: deflate_medium.c [hunk 1]
```diff
@@ -22,21 +22,15 @@ struct match {
...
-
 static int emit_match(deflate_state *s, struct match match) {
     int flush = 0;
+    int mflush = 0;
 
     /* matches that are not long enough we need to emit as literals */
     if (match.match_length < MIN_MATCH) {
         while (match.match_length) {
-            flush += tr_tally_lit(s, s->window[match.strstart]);
+            zng_tr_tally_lit(s, s->window[match.strstart], mflush);
```
Comment (Dead2) [inline deflate_medium.c]: These changes should not be part of this commit and likely not this PR.
Commits like the sweeping zng_ prefix changes should have zero effective change of code flow and performance.
The changes in this file might change both, and should get more scrutiny and not be hidden within a big commit.

Comment (nmoinvaz) [reply deflate_medium.c]: I removed them since renaming them zng_tr_tally_dist and zng_tr_tally_lit would conflict with functions that already carried the same name. I will add them back.

Comment (nmoinvaz) [reply deflate_medium.c]: I don't quite understand the purpose, of them; expect with the functions, you have additional function overhead for each call.

**Comment by nmoinvaz** [comment]
I have made the PR so it is only renaming of functions like you suggested.

File: deflate_medium.c (hunk 1 above)
Comment (nmoinvaz) [reply deflate_medium.c]: I can add a new PR for the removal of those extra functions after this PR, if it is necessary.

**Comment by nmoinvaz** [comment]
Looks like Travis CI stalled on one instance.

**Comment by Dead2** [comment]
Looks good to me, and it passes my tests.

**Comment by Dead2** [comment]
Thanks
